homeassistant/components/accuweather/weather.py

```diff
@@ -52,106 +51,127 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Add a AccuWeather weather entity from a config_entry."""
+    accuweather_data: AccuWeatherData = hass.data[DOMAIN][entry.entry_id]
 
-    coordinator: AccuWeatherDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([AccuWeatherEntity(accuweather_data)])
 
-    async_add_entities([AccuWeatherEntity(coordinator)])
 
-
-class AccuWeatherEntity(
-    SingleCoordinatorWeatherEntity[AccuWeatherDataUpdateCoordinator]
-):
+class AccuWeatherEntity(CoordinatorWeatherEntity):
```

Comment:
We can retain this coordinator type

Reply:
I don't understand, why do you want to use `SingleCoordinatorWeatherEntity`?

Reply:
I assumed the `CoordinatorWeatherEntity` also takes in a coordinator in the type `CoordinatorWeatherEntity[AccuWeatherDataUpdateCoordinator]` (but then the new coordinator)